public/app/features/transformers/utils.ts

```diff
@@ -45,10 +49,59 @@ export const getAllFieldNamesFromDataFrames = (frames: DataFrame[], withBaseFiel
   return names;
 };
 
+export const detectMixedQueryResults = (frames: DataFrame[]) =>
+  frames.some(({ fields }) => fields?.length > 0) && frames.some(({ fields }) => !fields?.length);
+
 export function useAllFieldNamesFromDataFrames(frames: DataFrame[], withBaseFieldNames = false): string[] {
   return useMemo(() => getAllFieldNamesFromDataFrames(frames, withBaseFieldNames), [frames, withBaseFieldNames]);
 }
 
+export const TransformerMissingFieldsMessage = () => {
+  return React.createElement(
+    FieldValidationMessage,
+    null,
+    t(
+      'transformers.query-validation-message',
+      'One or more queries failed. The transformation will only use fields from successful queries.'
+    )
+  );
+};
+
+type ExpandedTransformerUIProps<T> = TransformerUIProps<T> & { fieldNames: string[] };
+type DataFieldsErrorWrapperOptions = { withBaseFieldNames?: boolean };
+
+export const TIMEOUT = 300;
+
+export function DataFieldsErrorWrapper<T>(
```

Comment:
This wrapper is right on the edge of "deserves its own file" territory for me - I won't prevent the PR from being merged bc of it but if you find yourself being called in that direction, know I approve :D